lib/interfaces/ws.js

```diff
@@ -43,11 +43,22 @@ module.exports = function(app) {
       pathname: '/signalk/v1/stream',
       pingInterval: false
     });
-
+    
+    if ( app.securityPlugin )
+    {
+      app.securityPlugin.setupWS(primus)
+    }
+    
     primus.on('connection', function(spark) {
       debug(spark.id + " connected with params " + JSON.stringify(spark.query));
 
       var onChange = function(data) {
+        if ( app.securityPlugin && app.securityPlugin.verifyWS(spark) == false )
+        {
+          spark.end(undefined, { reconnect: true });
```

Comment:
I've only tested with wscat and WilhelmSK, it doesn't do anything there. I'll test with Primus...